src/tickit_devices/eiger/eiger_status.py

```diff
@@ -21,15 +23,21 @@ class State(Enum):
 class EigerStatus:
     """Stores the status parameters of the Eiger detector."""
 
-    state: State = field(default=State.NA)
-    errors: List[str] = field(default_factory=list)
-    th0_temp: float = field(default=24.5)
-    th0_humidity: float = field(default=0.2)
-    time: datetime = field(default=datetime.now())
-    dcu_buffer_free: float = field(default=0.5)
+    state: State = field(
+        default=State.NA,
+        metadata=rw_state(allowed_values=[state.value for state in State]),
+    )
+    errors: List[str] = field(default_factory=list, metadata=ro_str_list())
+    th0_temp: float = field(default=24.5, metadata=rw_float())
+    th0_humidity: float = field(default=0.2, metadata=rw_float())
+    time: datetime = field(default=datetime.now(), metadata=rw_datetime())
+    dcu_buffer_free: float = field(default=0.5, metadata=rw_float())
 
     def __getitem__(self, key: str) -> Any:  # noqa: D105
         f = {}
         for field_ in fields(self):
-            f[field_.name] = vars(self)[field_.name]
+            f[field_.name] = {
```

Comment:
Can we cache this value?

Or else calculate it without the intermediate object?

Reply:
https://github.com/dls-controls/tickit-devices/issues/67